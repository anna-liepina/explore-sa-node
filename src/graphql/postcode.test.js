describe('GraphQL: Postcode', () => {
    describe('query', () => {
        it(`fetching postcodes by exact postcodes`, async () => {
            const { data: { postcodeSearch: data } } = await query({
                query: `
                {
                    postcodeSearch(pattern: "E20 1AA") {
                        postcode
                        lat
                        lng
                        url
                    }
                }`
            });

            expect(data).toHaveLength(1);
            expect(data).toMatchSnapshot();
        });

        it(`fetching postcodes by wildcard`, async () => {
            const { data: { postcodeSearch: data } } = await query({
                query: `
                {
                    postcodeSearch(pattern: "E20") {
                        postcode
                        lat
                        lng
                        url
                    }
                }`
            });

            expect(data).toMatchSnapshot();
        });
    });
});
