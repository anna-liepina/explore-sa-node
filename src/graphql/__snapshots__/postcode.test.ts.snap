// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL: Postcode postcodeSearch [presets: perPage 3] by exact postcode 1`] = `
[
  {
    "lat": 51.547915089,
    "lng": -0.00743124,
    "postcode": "E20 1AA",
  },
]
`;

exports[`GraphQL: Postcode postcodeSearch [presets: perPage 3] by wildcard 1`] = `
[
  {
    "lat": 51.547915089,
    "lng": -0.00743124,
    "postcode": "E20 1AA",
  },
  {
    "lat": 51.548136,
    "lng": -0.007724,
    "postcode": "E20 1AB",
  },
  {
    "lat": 51.54796467,
    "lng": -0.007587,
    "postcode": "E20 1AD",
  },
]
`;

exports[`GraphQL: Postcode postcodeSearch [presets: perPage 3] by wildcard: 2nd page 1`] = `
[
  {
    "lat": 51.546844,
    "lng": -0.006872,
    "postcode": "E20 1AE",
  },
  {
    "lat": 51.547427816,
    "lng": -0.006803491,
    "postcode": "E20 1AF",
  },
  {
    "lat": 51.546690968,
    "lng": -0.006835695,
    "postcode": "E20 1AG",
  },
]
`;
